### PR TITLE
API: Instances index, show, and entrypoint 

### PR DESCRIPTION
**Pull request by rwsu**
Replaces #171.

Removed fields from show that are not ready to be displayed.

Updated date fields to new format.

Reference to deployment is now included in show.

Full resource representation is shown in top level collections.

### CODE REVIEW

**Comment by rwsu** [comment]
Here is a stack trace of the build error. 

malformed format string - %:
/home/rwsu/conductor/src/vendor/ruby/1.8/gems/i18n-0.6.0/lib/i18n/core_ext/string/interpolate.rb:101:in `interpolate_without_ruby_19_syntax'
 /home/rwsu/conductor/src/vendor/ruby/1.8/gems/i18n-0.6.0/lib/i18n/core_ext/string/interpolate.rb:101:in`send'
 /home/rwsu/conductor/src/vendor/ruby/1.8/gems/i18n-0.6.0/lib/i18n/core_ext/string/interpolate.rb:101:in `%'
 /home/rwsu/conductor/src/app/helpers/api_helper.rb:22:in`xmlschema_datetime'
 /home/rwsu/conductor/src/app/views/instances/_detail.xml.haml:12:in `_app_views_instances__detail_xml_haml__477859814_70277688006900'

**Comment by jistr** [comment]
Found the problem. The helper method is tested to work with DateTime, but I suppose Ruby classes aren't good enough for Rails :) I just looked at updated_at attribute, and it has a type ActiveSupport::TimeWithZone and their implementation of strftime doesn't seem to work in a standard way on Ruby 1.8. :rage4: Fortunately, they provide a to_datetime method on this thing, so this should fix it.

Don't change your code, I'm going to harden the helper (make it convert the parameter to DateTime) and submit a pull request for it. Then please review/merge it and rebase your pull request on top of that, and I hope it should be ok ;)

**Comment by jistr** [comment]
I hope that pull request #274 will fix it. Couldn't test it live on my machine because rbenv cannot compile Ruby 1.8 on the newer versions of GCC. Please ping me if the problem persists.

File: src/app/controllers/instances_controller.rb [hunk 1]
```diff
@@ -55,6 +56,7 @@ def show
         render :partial => @details_tab[:view] and return
       end
       format.json { render :json => @instance }
+      format.xml { render :show, :locals => { :instance => @instance } }
```
Comment (jistr) [inline src/app/controllers/instances_controller.rb:59]: Not totally sure, but I think the final consensus was that we shouldn't use :locals for passing from controllers to views when possible, and should use :locals for passing params to partials when possible [1]. It's a minor issue though.

https://lists.fedorahosted.org/pipermail/aeolus-devel/2012-August/thread.html#12054

**Comment by jistr** [comment]
Looks good to me and works for me. Conditional ACK with regard to the Ruby 1.8 thing.

**Pull request closed without merging**